lib/src/core_matchers.dart

```diff
@@ -14,7 +14,9 @@ class _Empty extends Matcher {
   const _Empty();
 
   @override
-  bool matches(Object? item, Map matchState) => (item as dynamic).isEmpty;
+  bool matches(Object? item, Map matchState) =>
+      // ignore: avoid_dynamic_calls
```

Comment:
It's too bad we don't have a better way to suppress these than the ignore. I filed https://github.com/dart-lang/linter/issues/4806

Reply:
Could this be replaced with `if (item is Iterable) return item.isEmpty; else if (item is String) return item.isEmpty;`? 

That would technically be breaking if someone defined an unknown object with an `isEmpty` getter. But the fact that it currently works is surprising to me. I thought the implementation relied on pattern match, not dynamic invocation

Reply:
I think supporting an `isEmpty` property on an arbitrary user defined class was an intentional feature and I'm not eager to try to remove it. It's not a design we'd carry forward to `checks`.